### ISSUE TITLE
Development option : fictitious position : it doesn't work for some devices (driver.setLocation)

### ISSUE THREAD

**Issue by Fanch-** [appium/appium#7151]
## The problem

I can use the driver.setLocation() method -org.openqa.selenium.html5-, with position allowed, and by enabling fictitious position (choosing appium settings app -for android version >= 6.0 it seems-) on Nexus 5X; it works fine (our app uses the GPS coordinates).

But on other device it doesn't work, my position is the current and not what I set in Location object.
For version < 6.0 I just tick fictitious position. On Nexus 6P (android 6.0) , I have to select the app, but appium settings isn't suggested.

## Environment

* Appium version 1.6.0
* java-client 4.1.0
* Application Mobile : real devices : Nexus 6P real device (6.0) , Nexus 5X (6.0, works), Htc One (5.0), Note3(5.0)

Is it flaky for mobile application or am I doing something wrong?

## Edit : 
Here the Samsung Note3 appium logs, the intent is correctly sent, same on every devices, so no java error returned : 

```
[HTTP] --> POST /wd/hub/session/68f8721e-7214-4600-9d0a-7b7583d4fac2/location {"location":{"altitude":0,"latitude":44.833328,"hCode":424751325,"class":"org.openqa.selenium.html5.Location","longitude":-0.56667}}
[MJSONWP] Calling AppiumDriver.setGeoLocation() with args: [{"altitude":0,"latitude":4...
[debug] [ADB] Getting connected devices...
[debug] [ADB] 12 device(s) connected
[debug] [ADB] Running 'path/sdk/platform-tools/adb' with args: ["-P",5037,"-s","059ab633","shell","am","startservice","-e","longitude",-0.56667,"-e","latitude",44.833328,"io.appium.settings/.LocationService"]
[MJSONWP] Responding to client with driver.setGeoLocation() result: "Starting service: Intent {...
[HTTP] <-- POST location 200
```

But in the phone logs I see : 
```
W/ActivityManager(31557): Unable to start service Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=io.appium.settings/.LocationService (has extras) } U=0: not found
```
Though global position and fictitious position (dev option) are allowed.

When it works (Nexus 5X -6.0-)
```
11-04 15:07:49.863 23420 23420 I Auth    : [AuthDelegateWrapper] Service intent: Intent { cmp=com.google.android.gms/.auth.account.authenticator.DefaultAuthDelegateService }.
11-04 15:07:49.864 23420 23420 I Auth    : [AuthDelegateWrapper] Service intent: Intent { cmp=com.google.android.gms/.auth.account.authenticator.DefaultAuthDelegateService }.
11-04 15:07:49.983 23420 23420 I Auth    : [AuthDelegateWrapper] Service intent: Intent { cmp=com.google.android.gms/.auth.account.authenticator.DefaultAuthDelegateService }.
11-04 14:29:24.286 23420 23420 I GCoreUlr: Starting service, intent=Intent { act=com.google.android.location.reporting.ACTION_UPDATE_WORLD cmp=com.google.android.gms/com.google.android.location.reporting.service.DispatchingService (has extras) }, extras=Bundle[{receiverAction=android.location.PROVIDERS_CHANGED}]
11-04 14:29:24.418 23420 27975 I GCoreUlr: WorldUpdater received intent Intent { act=com.google.android.location.reporting.ACTION_UPDATE_WORLD cmp=com.google.android.gms/com.google.android.location.reporting.service.DispatchingService (has extras) } with receiverAction android.location.PROVIDERS_CHANGED
```

On Nexus 6P -6.0- (where I can't choose the appium settings app for fictitious position in dev options) : 
```
11-04 14:36:00.900 29252 29252 I Auth    : [AuthDelegateWrapper] Service intent: Intent { cmp=com.google.android.gms/.auth.account.authenticator.DefaultAuthDelegateService }.
11-04 14:36:00.901 29252 29252 I Auth    : [AuthDelegateWrapper] Service intent: Intent { cmp=com.google.android.gms/.auth.account.authenticator.DefaultAuthDelegateService }.
11-04 14:36:00.956 29252 29252 I Auth    : [AuthDelegateWrapper] Service intent: Intent { cmp=com.google.android.gms/.auth.account.authenticator.DefaultAuthDelegateService }.
11-04 14:36:04.096   936  3278 W ActivityManager: Unable to start service Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=io.appium.settings/.LocationService (has extras) } U=0: not found
11-04 14:37:05.948   936  3088 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.HOME] flg=0x10200000 cmp=com.google.android.googlequicksearchbox/com.google.android.launcher.GEL (has extras)} from uid 1000 on display 0
11-04 14:37:06.572 25414 25414 I GEL     : handleIntent(Intent { act=android.intent.action.MAIN cat=[android.intent.category.HOME] flg=0x10200000 cmp=com.google.android.googlequicksearchbox/com.google.android.launcher.GEL (has extras) })
11-04 14:38:28.241   936  4056 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=fr.leboncoin.delivery/fr.leboncoin.ui.activities.SwitcherActivity (has extras)} from uid 2000 on display 0
11-04 14:38:29.342   936   954 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.DEFAULT] flg=0x10008000 cmp=fr.leboncoin.delivery/fr.leboncoin.ui.activities.SplashScreenActivity} from uid 10276 on display 0
11-04 14:38:32.661   936  4479 W ActivityManager: Unable to start service Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=io.appium.settings/.LocationService (has extras) } U=0: not found
```

So also : 
```
Unable to start service Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=io.appium.settings/.LocationService (has extras) } U=0: not found
```

To reproduce it I guess setting a location or launching the corresponding adb command on these devices should reproduce the error. 

Java code :

```  
public static final double AQUITAINE_LATITUDE = 44.833328;
public static final double AQUITAINE_LONGITUDE = -0.56667;
public static Location getGpsLocationAquitaine(){
    return new Location(AQUITAINE_LATITUDE, AQUITAINE_LONGITUDE, 0.0);
 }
//then 
    driver.setLocation(LocalisationFactory.getGpsLocationAquitaine());
```

**Comment by dpgraham** [comment]
Hi @Fanch- , what is the error that you're experiencing? Are Java exceptions being thrown when you call driver.setLocation?

Can you provide us with the Appium server logs as well as any Java error logs and can you give us the steps to reproduce the error.

**Comment by Fanch-** [comment]
@dpgraham , Here the logs, sorry I forgot them**.
Do you see any further information needed?

**Comment by dpgraham** [comment]
@imurchie do you mind having a look at this one. I think it's a bit over my head.

**Comment by WitoldKijKijowieckiMobica** [comment]
any progress in that matter?

**Comment by ndinoia** [comment]
I'm having issues with setting fake location as well. Tried on both emulators and real devices with no luck. `driver.setLocation` will return OK but afterwards when I call `driver.location()` I get a `SocketTimeoutException` (custom timeout of 5 minutes). I checked manually after the tests fail and google maps on the device/emu does not show the fake location had been applied.

Tested on Nexus 6P emulator with android 7.1.1
And on a real Google Pixel / Android 8

Appium v1.8.0
java-client v6.0.0-BETA5
